src/EditorFeatures/Test/Extensions/SourceTextContainerExtensionsTests.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.CodeAnalysis.Text;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.Extensions
+{
+    public class SourceTextContainerExtensionsTests
+    {
+        [Fact]
+        public void GetBufferTextFromNonTextContainerThrows()
+        {
+            var containerMock = new Mock<SourceTextContainer>();
+            Assert.Throws<ArgumentException>(containerMock.Object.GetTextBuffer);
```

Comment:
The intention was to test that the GetTextBuffer extension method threw an argument exception when provided with a SourceTextContainer that was not a TextBufferContainer or derived type. Would it have been better to have written it as 
```Assert.Throws<ArgumentException>(() => Microsoft.CodeAnalysis.Text.Extensions.GetTextBuffer(containerMock.Object));```

Or something similar?

I did also look at asserting on the ArgumentException message to make sure that the update resource text was returned however as this also returns the parameter name as part of the message I wasn't confident that this wouldn't make the test too brittle because someone could easily break the test just by changing the parameter name.